cpp/src/arrow/dataset/scanner_internal.h

```diff
@@ -185,6 +185,12 @@ inline Result<ScanTaskIterator> GetScanTaskIterator(
   auto fn = [options](std::shared_ptr<Fragment> fragment) -> Result<ScanTaskIterator> {
     ARROW_ASSIGN_OR_RAISE(auto scan_task_it, fragment->Scan(options));
 
+    if (fragment->type_name() == "skyhook") {
```

Comment:
Is Skyhook actually doing projection?  I think filtering is fine but if projection is done early at the format level then I believe that may cause issues when executing datasets that come from a variety of sources (e.g. unions and joins).

Reply:
Or rather, we shouldn't skip filter and projection here right? Even if it's pushed down (as can be done already in Parquet) we should keep the rest of the pipeline uniform. (At least, projection should be cheap - it may be understandable to skip filtering entirely if the format guarantees that filtering is completely pushed down.)

Reply:
Yes, filtering (although is quite fast) incurs some overhead. Maybe we can have a flag in `ScanOptions` to switch on/off client-side filtering ?